### PR TITLE
Task/pipeline

### PR DESCRIPTION
**Pull request by mmelograno**
- Implement pipeline for Pluggable and Redis.

### CODE REVIEW

File: redis-wrapper/src/main/java/redis/RedisPipeline.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    private final JedisPool _jedisPool;
+
+
+    public RedisPipeline(JedisPool jedisPool, String prefix) {
+        _jedisPool = jedisPool;
+        _prefix = prefix;
+        try (Jedis jedis = jedisPool.getResource()) {
+            _pipelined = jedis.pipelined();
+        } catch (Exception ex) {
+            System.out.println("err " + ex.getMessage());
```
Comment (mmelograno) [inline redis-wrapper/src/main/java/redis/RedisPipeline.java:23]: it should throws RedisException

File: redis-wrapper/src/main/java/redis/RedisPipeline.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    public void delete(List<String> keys){
+        if(keys == null || keys.isEmpty()){
+            return ;
+        }
+        try (Jedis jedis = _jedisPool.getResource()) {
+            keys = keys.stream().map(key -> buildKeyWithPrefix(key)).collect(Collectors.toList());
+
+            jedis.del(keys.toArray(new String[keys.size()]));
+        } catch (Exception ex) {
+            throw ex;
```
Comment (mmelograno) [inline redis-wrapper/src/main/java/redis/RedisPipeline.java:50]: it should throws RedisException

Comment (nmayorsplit) [reply redis-wrapper/src/main/java/redis/RedisPipeline.java]: Changed it.

File: redis-wrapper/src/main/java/redis/RedisPipeline.java (hunk 1 above)
Comment (nmayorsplit) [reply redis-wrapper/src/main/java/redis/RedisPipeline.java]: Changed it.

File: redis-wrapper/src/main/java/redis/RedisPipeline.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        _jedisPool = jedisPool;
+        _prefix = prefix;
+        try (Jedis jedis = _jedisPool.getResource()) {
+            _pipelined = jedis.pipelined();
+        } catch (Exception ex) {
+            new RedisException(ex.getMessage());
+        }
+    }
+
+    /* package private */ String buildKeyWithPrefix(String key) {
```
Comment (mmelograno) [inline redis-wrapper/src/main/java/redis/RedisPipeline.java:30]: this method is used in several places, can we refactor to have in some common place?

File: redis-wrapper/src/main/java/redis/RedisPipeline.java [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        if (!key.startsWith(_prefix)) {
```
Comment (mmelograno) [inline redis-wrapper/src/main/java/redis/RedisPipeline.java:31]: this checking is an extra thing that should be removed, if for some reason two prefix are concatenated, definitely we have a bug in the code

File: redis-wrapper/src/main/java/redis/RedisPipeline.java (hunk 3 above)
Comment (nmayorsplit) [reply redis-wrapper/src/main/java/redis/RedisPipeline.java:30]: I'm going to work on this using another branch.

File: redis-wrapper/src/main/java/redis/RedisPipeline.java (hunk 4 above)
Comment (nmayorsplit) [reply redis-wrapper/src/main/java/redis/RedisPipeline.java:31]: The same before comment.

File: client/src/main/java/io/split/storages/pluggable/domain/UserPipelineWrapper.java [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+        _pipeline = pipeline;
+    }
+
+    @Override
+    public List<Result> exec() throws Exception {
+        try{
+            return _pipeline.exec();
+        } catch (Exception e) {
+            _logger.warn("Exception calling Pipeline exec", e);
+            throw new Exception(e.getMessage());
```
Comment (mmelograno) [inline client/src/main/java/io/split/storages/pluggable/domain/UserPipelineWrapper.java:27]: you can return an empty list

Comment (nmayorsplit) [reply client/src/main/java/io/split/storages/pluggable/domain/UserPipelineWrapper.java]: I added the return.

**Pull request merged**